src/NodaTime.Web/Startup.cs

```diff
@@ -80,6 +84,43 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 ContentTypeProvider = new FileExtensionContentTypeProvider
                 {
                     Mappings = { [".nzd"] = "application/octet-stream" }
+                },
+                OnPrepareResponse = (context) => {
+                    // Set the Cache-Control header for static content.
+                    var headers = new ResponseHeaders(context.Context.Response.Headers);
+
+                    // All static content is cacheable by intermediates.
+                    headers.CacheControl = new CacheControlHeaderValue()
```

Comment:
Removed this (it really only affects /tzdb/).

It would be nice to send an ETag for text/html content (like Markdown) and allow the browser to revalidate the cached version cheaply, but I'm wary of allowing the browser to avoid the round-trip entirely.

In particular, since ASP.NET doesn't validate the `?v=` parameter is actually correct, you can get a nasty problem if you allow an old referencing page to be used without revalidation:

1. Browser fetches `/foo/` (HTML) and then `/foo.css?v=1` (JS). The first has Cache-Control: max-age=5 minutes, the latter Cache-Control: max-age=1 year.
2. The site is updated. `/foo/` now uses `/foo.css?v=2`.
2. The user navigates back to `/foo/`. The browser reuses the HTML resource without revalidating it, but refetches `/foo.css?v=1` (because it's dropped out of the cache). This actually returns the _new_ content (I'd consider this a bug), and the browser permanently caches the new CSS with the old HTML content. (Alternatively: the server sensibly returns 404, and the page is also broken.)